crates/transpiler/src/passes/sabre/route.rs

```diff
@@ -190,8 +190,15 @@ impl RoutingResult<'_> {
             for swap in item.initial_swaps() {
                 apply_swap(swap, &mut layout, &mut dag)?;
             }
-            let index = self.sabre.dag[item.node].index;
-            let NodeType::Operation(inst) = &self.dag[index] else {
+            let Some((head, rest)) = self.sabre.dag[item.node].indices.split_first() else {
+                // In theory this should never actually trigger: we're in a state where there was a
+                // Sabre routing-related interaction with no DAG node backing it.  That said, we do
+                // sometimes construct interaction graphs in this state (via
+                // [SabreDAG::only_interactions]), and either way, there's a meaningful way to
+                // handle it safely.
+                continue;
+            };
```

Comment:
I actually found this syntax a bit confusing here. I couldn't understand why you would be skipping this iteration if there were routable nodes listed until I finally saw the `else` at the end of the line,

I know you're fond of this shorthand because it eliminates an indent but it might be a bit easier to read if you did something like:

```rust
let split_iter = self.sabre.dag[item.node].indices.split_first();
if split_iter().is_none() {
    continue;
}
let (head, rest) = split_iter.unwrap();
```
instead. Not a blocker or anything, just a small nit.